throttler/throttler.go

```diff
@@ -0,0 +1,78 @@
+package throttler
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+	"time"
+)
+
+//ThrottledAction is a common interface for actions that can be executed
+type ThrottledAction interface {
+	Execute()
+}
```

Comment:
In fact, I think this could be a Runner decorator. We discussed this architecture with @ferrastas the other day.